### PR TITLE
docs(CONTRIBUTING): update contributing according to latest xcode

### PR DESCRIPTION
**Pull request by saurabhdaware**
## Changes

- Added some notes to CONTRIBUTING file to help simplify the process and reduce any confusions
- Minor tweaks in CONTRIBUTING
- Added my github username to CODEOWNERS 🕺🏻

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,13 +38,13 @@ These steps should get you up and started for local development setup. Please en
 
 - Install [Xcode](https://reactnative.dev/docs/environment-setup#xcode).
 
-  > **Note:**
+  > **Note**
```
Comment (saurabhdaware) [inline CONTRIBUTING.md:41]: It will render with that GitHub "Note" UI with this

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -38,13 +38,13 @@ These steps should get you up and started for local development setup. Please en
...
   >
   > Sometimes it can take a very long time for Xcode to install. Check [here](https://apple.stackexchange.com/questions/427640/mac-app-store-xcode-download-stuck-at-installing) for troubleshooting.
 
-- Install an iOS simulator in Xcode: Xcode > Preferences > Components.
+- Install an iOS 13 simulator in Xcode: Xcode > Preferences > Platforms.
```
Comment (saurabhdaware) [inline CONTRIBUTING.md:45]: xcode installed ios 16 by default in my machine and the storybook react native command was failing saying it cannot find "iPhone 13".

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: CONTRIBUTING.md (hunk 2 above)
Comment (saurabhdaware) [reply CONTRIBUTING.md:45]: I can't see "Components" tab in my preferences. I have "Platforms" from where I can install ios or any other simulators and I am on latest xcode.

**Review by saurabhdaware — COMMENTED** [review]
Review Helpers

**Pull request merged**